src/dynamic_loader.cpp

```diff
@@ -48,7 +48,7 @@ struct dynamic_loader_impl
 #pragma GCC diagnostic ignored "-Wignored-attributes"
 #endif
     dynamic_loader_impl(const fs::path& p, std::shared_ptr<tmp_dir> t = nullptr)
-        : handle(dlopen(p.string().c_str(), RTLD_LAZY),
+        : handle(dlopen(p.string().c_str(), RTLD_GLOBAL | RTLD_NOW),
```

Comment:
Can it work with `RTLD_LAZY` ?  Not sure but having it `RTLD_NOW` would force it to resolve all the symbols.

Reply:
It doesnt work. It seems like `RTLD_LAZY` would not be as useful since we need to load everything right away. We can add some kind of lazy flag in the future if need to load them lazily.

Reply:
> It doesnt work. 

Curions, why it wouldn't work. Lazy would load on need basis. If it requires something it would load it. `RTLD_GLOBAL` would load everything.  I am trying to figure out where it would break for `LAZY`.

Reply:
> RTLD_GLOBAL would load everything.

No, `RTLD_GLOBAL` makes the symbols available for subsequently loaded libraries to use. 

>  I am trying to figure out where it would break for LAZY.

With how python loads the libraries it causes undefined references from missing dependencies. The only way to it to work is to use `RTLD_GLOBAL | RTLD_NOW`. Of course, you can try to see if you can find another way to make it work with `RTLD_LAZY`.